src/main/java/uk/gov/companieshouse/itemhandler/service/EmailService.java

```diff
@@ -103,56 +51,34 @@ public EmailService(
      */
     public void sendOrderConfirmation(DeliverableItemGroup itemGroup) {
         try {
-            if (ITEM_KIND_CERTIFIED_COPY.equals(itemGroup.getKind())) {
-                final OrderConfirmationAndEmail orderConfirmationAndEmail = buildOrderConfirmationAndEmail(itemGroup.getOrder());
-                final OrderConfirmation confirmation = orderConfirmationAndEmail.confirmation;
-                final EmailSend email = orderConfirmationAndEmail.email;
-                email.setEmailAddress(TOKEN_EMAIL_ADDRESS);
-                email.setMessageId(UUID.randomUUID().toString());
-                email.setData(objectMapper.writeValueAsString(confirmation));
-                email.setCreatedAt(LocalDateTime.now().toString());
-
-                String orderReference = confirmation.getOrderReferenceNumber();
-                LoggingUtils.logWithOrderReference("Sending confirmation email for order", orderReference);
-                emailSendProducer.sendMessage(email, orderReference);
-            } else {
-                EmailMetadata<CertificateEmailData> emailMetadata = confirmationMapperFactory.getCertificateMapper().map(itemGroup);
-                EmailSend emailSend = new EmailSend();
-                emailSend.setAppId(emailMetadata.getAppId());
-                emailSend.setMessageType(emailMetadata.getMessageType());
-                emailSend.setData(objectMapper.writeValueAsString(emailMetadata.getEmailData()));
-                emailSend.setEmailAddress(TOKEN_EMAIL_ADDRESS);
-                emailSend.setMessageId(UUID.randomUUID().toString());
-                emailSend.setCreatedAt(LocalDateTime.now().toString());
+            EmailSend emailSend = createEmailSendBasedOnKind(itemGroup);
+            emailSend.setEmailAddress(TOKEN_EMAIL_ADDRESS);
```

Comment:
Lines 55-57 break encapsulation; mapping logic has leaked out of your mapping method into client code.

Reply:
Resolved in latest commit.